From 3511defbf42a3f8cf0fd74b8ae19869406f18b6f Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Thu, 2 Feb 2017 11:10:08 -0800
Subject: [PATCH 075/131] soapy: added device::find() support

Was missing from the implementation.
Now devices with a label show up nicely in GQRX
---
 lib/device.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lib/device.cc b/lib/device.cc
index 3ce40fa..d75d6ad 100644
--- a/lib/device.cc
+++ b/lib/device.cc
@@ -78,6 +78,10 @@
 #include <airspy_source_c.h>
 #endif
 
+#ifdef ENABLE_SOAPY
+#include <soapy_source_c.h>
+#endif
+
 #ifdef ENABLE_REDPITAYA
 #include <redpitaya_source_c.h>
 #endif
@@ -178,6 +182,10 @@ devices_t device::find(const device_t &hint)
   BOOST_FOREACH( std::string dev, airspy_source_c::get_devices() )
     devices.push_back( device_t(dev) );
 #endif
+#ifdef ENABLE_SOAPY
+  BOOST_FOREACH( std::string dev, soapy_source_c::get_devices() )
+    devices.push_back( device_t(dev) );
+#endif
 
   /* software-only sources should be appended at the very end,
    * hopefully resulting in hardware sources to be shown first
-- 
2.20.1

